adoc/extensions/sycl_khr_group_interface.adoc

```diff
@@ -0,0 +1,479 @@
+[[sec:khr-group-interface]]
+= SYCL_KHR_GROUP_INTERFACE
+
+This extension provides an alternative interface for groups of work-items
+(including work-groups, sub-groups, and individual work-items) that is simpler
+and less verbose than the interface provided by `sycl::group` and
+`sycl::sub_group` in SYCL 2020.
+
+[[sec:khr-group-interface-dependencies]]
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+Some features of this extension are only available when an implementation
+provides `<mdspan>`.
+
+[[sec:khr-group-interface-feature-test]]
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_GROUP_INTERFACE# to one of the values defined in the table
+below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+[[sec:khr-group-interface-common]]
+== Common group interface
+
+The `khr::work_group`, `khr::sub_group` and `khr::work_item` objects defined by
+this extension all implement a common set of operations.
+
+[source,role=synopsis]
+----
+
+namespace sycl {
+namespace khr {
+
+class GroupT {
+
+ public:
+  using id_type = /* ... */;
+  using linear_id_type = /* ... */;
+  using range_type = /* ... */;
+#if defined(__cpp_lib_mdspan)
+  using extents_type = /* ... */;
+  using index_type = typename extents_type::index_type; // exposition only
+  using rank_type = typename extents_type::rank_type; // exposition only
```

Comment:
I declared these as exposition only because we don't want them to be exposed by classes implementing the group interface, but they simplified the synopsis.  The alternative is writing `extents_type::index_type` everywhere it appears; I've made this change in dd2110a6f71f9ca00d688296dc6f7f3dcaf0bfb8.

 I don't want to assume `std::dextents`, because we already know that future group types will use different `extents_type`s, and I'm worried we'll introduce a specification bug by forgetting to revisit this.